src/library_webgpu.js

```diff
@@ -2408,9 +2408,6 @@ var LibraryWebGPU = {
     var selectorPtr = {{{ makeGetValue('descriptorFromCanvasHTMLSelector', C_STRUCTS.WGPUSurfaceDescriptorFromCanvasHTMLSelector.selector, '*') }}};
     {{{ gpu.makeCheck('selectorPtr') }}}
     var canvas = findCanvasEventTarget(selectorPtr);
-#if ASSERTIONS
-    assert(canvas instanceof HTMLCanvasElement);
```

Comment:
What would this be an instance of in the off-screen canvas case?

Reply:
Ah yes, I should have linked to a bug! This was reported in #17776. You can put arbitrary objects into the specialHTMLTargets object which lets you get an OffscreenCanvas from findCanvasEventTarget. Technically this sounds like a type error, but it's otherwise perfectly valid because getContext is valid on it.